### PR TITLE
Implemented LongtermFFTVisualizer base

### LINKED ISSUE #210: Missing LongtermFFTVisualizer
**Actual Behaviour.**  
Missing LongtermFFTVisualizer

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by jddeep**
Fixes #210 

**Changes**: Implemented LongtermFFTVisualizer base.

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

### CODE REVIEW

File: app/src/main/java/io/neurolab/main/output/visual/LongtermFFTVisualizer.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    public void run() {
+        if (nfbServer.getNumSamples() < 1)
+            return;
+
+        int numSamples = nfbServer.getNumSamples();
+        for (int i = 0; i < numPlaybackModes; i++) {
+            sampleCounter[i] += numSamples;
+        }
+
+
```
Comment (suresh-gandhi) [inline app/src/main/java/io/neurolab/main/output/visual/LongtermFFTVisualizer.java:79]: Remove an extra space of line from here.

File: app/src/main/java/io/neurolab/main/output/visual/LongtermFFTVisualizer.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        {
+
+            for (int c = 0; c < numChannels; c++)
```
Comment (suresh-gandhi) [inline app/src/main/java/io/neurolab/main/output/visual/LongtermFFTVisualizer.java]: here also

File: app/src/main/java/io/neurolab/main/output/visual/LongtermFFTVisualizer.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                            minFFT = Math.min(minFFT, fftData.getCurrentFFTs()[i][f]);
+
+                            int colorVal = (int) (((fftData.getCurrentFFTs()[i][f] - minFFT) / maxFFT) * 255d);
+
+                            if (colorVal > 255)
+                                colorVal = 255;
+                            else if (colorVal < 0)
+                                colorVal = 0;
+
+
```
Comment (suresh-gandhi) [inline app/src/main/java/io/neurolab/main/output/visual/LongtermFFTVisualizer.java]: here also.

File: app/src/main/java/io/neurolab/main/output/visual/LongtermFFTVisualizer.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+                        }
+                    }
+                    if (this.fbSettings != null) {
+                        int colorVal = (int) (this.fbSettings.getCurrentFeedback() * 1024d);
+
+                        if (colorVal > 255)
+                            colorVal = 255;
+                        else if (colorVal < 0)
+                            colorVal = 0;
+
```
Comment (suresh-gandhi) [inline app/src/main/java/io/neurolab/main/output/visual/LongtermFFTVisualizer.java]: here

File: app/src/main/java/io/neurolab/main/output/visual/LongtermFFTVisualizer.java [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+                    }
+                }
+            if (this.fbSettings != null) {
+                int colorVal = (int) (this.fbSettings.getCurrentFeedback() * 1024d);
+
+                if (colorVal > 255)
+                    colorVal = 255;
+                else if (colorVal < 0)
+                    colorVal = 0;
+
```
Comment (suresh-gandhi) [inline app/src/main/java/io/neurolab/main/output/visual/LongtermFFTVisualizer.java]: here

File: app/src/main/java/io/neurolab/main/output/visual/LongtermFFTVisualizer.java [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+        if (playbackSpeed >= numPlaybackModes)
+            playbackSpeed = 0;
+    }
+
+    @Override
+    public void init() {
+        // initialize the Visualizer
+    }
+
+    int[] sampleCounter;
```
Comment (suresh-gandhi) [inline app/src/main/java/io/neurolab/main/output/visual/LongtermFFTVisualizer.java]: Rearrange the code - variables top and methods below

**Review by suresh-gandhi — CHANGES_REQUESTED** [review]
@jddeep Everything is good to go. Just needs a bit of reformatting and rearranging.

**Comment by jddeep** [comment]
@suresh-gandhi Done the formatting changes 👍

**Review by mariobehling — APPROVED** [review]

**Pull request merged**